tap_egencia/streams.py

```diff
@@ -2,27 +2,72 @@
 
 from __future__ import annotations
 
+import requests
+
 from pathlib import Path
 
 from singer_sdk import typing as th  # JSON Schema typing helpers
 
+from singer_sdk.helpers.jsonpath import extract_jsonpath
+
 from tap_egencia.client import egenciaStream
 
+from singer_sdk.typing import PropertiesList, Property, StringType
+
+from typing import Any, Dict, Iterable, Optional
+
+
+from tap_egencia.schemas.get_transactions import metadataObject, transactionsObject
+
 SCHEMAS_DIR = Path(__file__).parent / Path("./schemas")
 
 
-# TODO: - Rename Stream to appropriately capture the dataStream and path. 
-#       - Override `UsersStream` and `GroupsStream` with your own stream definition.
-class UsersStream(egenciaStream):
-    """Define custom stream."""
-
-    primary_keys = ["id"]
-    path = '/comments'
-    name = "comments"
-    schema = th.PropertiesList(
-        th.Property("postId", th.IntegerType),
-        th.Property("id", th.IntegerType),
-        th.Property("name", th.StringType),
-        th.Property("email", th.StringType),
-        th.Property("body", th.StringType),
+class TransactionsStream(egenciaStream):
+    """Define reporting/transactions stream."""
+
+    name = "transactions-api"
+    schema = PropertiesList(
+        Property("report_id", StringType),
+        Property("metadata", metadataObject),
+        Property("transactions", transactionsObject),
     ).to_dict()
+    authenticator = None
+
+    def parse_response(self, response: requests.Response) -> Iterable[dict]:
+        return [] if response.status_code == 400 else super().parse_response(response)
```

Comment:
Should we check for other error codes here as well? 401? 403? 422? 500?

Reply:
sure, I can have it check for those as well.

Reply:
@Drakmyth Updated